src/logic/events/TrackpadEvents.swift

```diff
@@ -0,0 +1,113 @@
+import M5MultitouchSupport
+
+//TODO: patch M5MultitouchSupport pod to make it not to crash after sleep. See https://github.com/mhuusko5/M5MultitouchSupport/issues/1
+//TODO: App Expose activates if swipe down with 3-fingers right after AltTab activation. There is no issue if wait for some time.
+//TODO: enable/disable trackpad usage in settings
+class TrackpadEvents {
+    //TODO: Should we add a sensetivity setting instead of these magic numbers?
+    private static let accVelXThreshold: Float = 10
+    private static let accVelYThreshold: Float = 15
+    
+    private static var listener: M5MultitouchListener?
+
+    static func addSwipeListener() {
+        var accVelX: Float = 0
+        var accVelY: Float = 0
+        // We need to track an UI activation via gesture to not to hide UI on trackpad events for other activations.
+        var activated: Bool = false
+
+        listener = M5MultitouchManager.shared().addListener {event in
+            if event == nil {
+                return
+            }
+
+            let touches = event!.touches as! [M5MultitouchTouch]
+
+            // We don't care about non-3-fingers swipes.
+            if touches.count != 3 {
+                // Except when we already started a gesture, so we need to end it.
+                if activated {
+                    if App.app.appIsBeingUsed {
+                        DispatchQueue.main.async {
+                            App.app.focusTarget()
+                            App.app.hideUi()
+                        }
+                    }
+                    accVelX = 0
+                    accVelY = 0
+                    activated = false
+                }
+                return
+            }
+
+            let velocity = swipeVelocity(touches: touches)
+            // We don't care about gestures other than horizontal or vertical swipes.
+            if velocity == nil {
+                return
+            }
+
+            accVelX += velocity!.x
+            accVelY += velocity!.y
+            // Not enough swiping.
+            if abs(accVelX) < accVelXThreshold && abs(accVelY) < accVelYThreshold {
+                return
+            }
+
+            let isHorizontal = abs(velocity!.x) > abs(velocity!.y)
+            if App.app.appIsBeingUsed {
+                let direction: Direction = isHorizontal
+                    ? accVelX < 0 ? .left : .right
+                    : accVelY < 0 ? .down : .up
+                DispatchQueue.main.async { App.app.cycleSelection(direction) }
+            } else {
+                if isHorizontal {
+                    DispatchQueue.main.async { App.app.showUi() }
+                    activated = true
```

Comment:
Do we need to track this bool actually? Maybe `App.app.appIsBeingUsed` is enough?

Shouldn't the user be able to open AltTab with alt+tab, but then 3-fingers swipe left or right?

Reply:
We need to track that the UI was triggered by swiping because end of the swipe (number of fingers not equal to 3 in our case) hides the UI. If a user opens the UI from menu or by shortcut, every mouse move except 3-finger swipe will close the UI.

Reply:
Maybe the following logic could work?

We track that we reached 3 fingers either:

* Before AltTab was opened with the trackpad gesture
* After it was opened with the keyboard shortcut

From that point on, any < 3 fingers gesture means we should focus the window and close UI.

Reply:
We could track what caused the UI to be shown by adding a parameter to `showUI` method and storing it.